nordicsemi/dfu/dfu_transport_ble.py

```diff
@@ -231,7 +231,9 @@ def close(self):
             raise IllegalStateException('DFU Adapter is already closed')
         super(DfuTransportBle, self).close()
         self.dfu_adapter.close()
-        self.dfu_adapter = None
+        if self.dfu_adapter is not None:
```

Comment:
I don't see the reason for this change. You want dfu_adapter to be None, so there should be no need to check the value first.